src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java

```diff
@@ -623,6 +624,14 @@ private static void configureJdbcDefaults(OracleDataSource oracleDataSource) {
       OracleConnection.CONNECTION_PROPERTY_DEFAULT_LOB_PREFETCH_SIZE,
       "1048576");
 
+    // Disable out-of-band breaks. JDBC sends one of these when a statement
+    // timeout expires. When enabled, the database doesn't seem to respond by
+    // cancelling the statement execution. Disable this so that statement
+    // timeouts will work correctly.
+    setPropertyIfAbsent(oracleDataSource,
+      OracleConnection.CONNECTION_PROPERTY_THIN_NET_DISABLE_OUT_OF_BAND_BREAK,
```

Comment:
Out of band break should provide a more reliable way to cancel the current RPC. Disabling it doesn't look right to me.

Reply:
OOB breaks are definitely faster. Without OOB, there's a multi-second lag between JDBC sending a break and the database responding with ORA-01013. However, if we are supporting the 18.x database release, then I think we should have OOB disabled by default.

It is only starting in 19.x that the database is able to perform an OOB check with the client. If the check fails, then the client knows to disable OOB. But the 18.x database did not support this check, so if OOB is not working then it won't be detected and statement timeouts are effectively unsupported.

So I think the safest choice is to disable OOB by default, and expose this connection property so that applications may enable it if their database is 19.x or newer. Once 18.x databases are no longer supported, I think it would be safe to change the default behavior so that OOB is enabled by default, and let the driver disabled it only if the check fails.

An alternative is to leave OOB enabled by default, and have the R2DBC Option to disable it for older databases. This is great if your system connects to a newer database: You get fast timeouts by default. This is awful if you connect to an older database: You're system hangs and you have no idea why. You're calling setStatementTimeout(Duration) correctly, but the driver's implementation is totally incorrect. You'd need to configure a non-default setting in order for the driver to operate correctly. I don't think a default setting should be one that is incompatible with a database version that we support.

Reply:
Latest commit adds support to configure this connection property as a ConnectionFactory Option. Whether we decide to enable or disable OOB by default, an Option to override the default should be supported.

Reply:
It's good to have the option to enable or disable out of band break. By default out of band break should be enabled like it is in the thin-driver.

Reply:
OOB is left enabled by default in the latest commit.